examples/3d/fog.rs

```diff
@@ -0,0 +1,310 @@
+                end: 20.0,
+            },
+        };
+    }
+
+    if keycode.pressed(KeyCode::Key2) {
+        fog.mode = match fog.mode {
+            FogMode::Exponential { density } => FogMode::Exponential { density },
+            FogMode::ExponentialSquared { density } => FogMode::Exponential { density },
+            _ => FogMode::Exponential { density: 0.07 },
+        };
+    }
+
+    if keycode.pressed(KeyCode::Key3) {
+        fog.mode = match fog.mode {
+            FogMode::Exponential { density } => FogMode::ExponentialSquared { density },
+            FogMode::ExponentialSquared { density } => FogMode::ExponentialSquared { density },
+            _ => FogMode::ExponentialSquared { density: 0.07 },
+        };
+    }
+
+    // Linear Fog Controls
+    if let FogMode::Linear {
+        ref mut start,
+        ref mut end,
+    } = &mut fog.mode
+    {
+        text.sections[0]
+            .value
+            .push_str(format!("\nA / S - Move Start Distance\nZ / X - Move End Distance").as_str());
+
+        if keycode.pressed(KeyCode::A) {
+            *start -= delta * 3.0;
+        }
+        if keycode.pressed(KeyCode::S) {
+            *start += delta * 3.0;
+        }
+        if keycode.pressed(KeyCode::Z) {
+            *end -= delta * 3.0;
+        }
+        if keycode.pressed(KeyCode::X) {
+            *end += delta * 3.0;
+        }
+    }
+
+    // Exponential Fog Controls
+    if let FogMode::Exponential { ref mut density } = &mut fog.mode {
+        text.sections[0]
+            .value
+            .push_str(format!("\nA / S - Change Density").as_str());
+
+        if keycode.pressed(KeyCode::A) {
+            *density -= delta * 0.5 * *density;
+            if *density < 0.0 {
+                *density = 0.0
+            }
+        }
+        if keycode.pressed(KeyCode::S) {
+            *density += delta * 0.5 * *density;
+        }
+    }
+
+    // ExponentialSquared Fog Controls
+    if let FogMode::ExponentialSquared { ref mut density } = &mut fog.mode {
+        text.sections[0]
+            .value
+            .push_str(format!("\nA / S - Change Density").as_str());
+
+        if keycode.pressed(KeyCode::A) {
+            *density -= delta * 0.5 * *density;
+            if *density < 0.0 {
+                *density = 0.0
+            }
+        }
+        if keycode.pressed(KeyCode::S) {
+            *density += delta * 0.5 * *density;
+        }
+    }
+
+    // RGBA Controls
+    text.sections[0]
+        .value
+        .push_str(format!("\n\n- / = - Red\n[ / ] - Green\n; / ' - Blue\n. / ? - Alpha").as_str());
+
+    if keycode.pressed(KeyCode::Minus) {
+        let r = (fog.color.r() - 0.1 * delta).max(0.0);
+        fog.color.set_r(r);
+    }
+
+    if keycode.pressed(KeyCode::Equals) {
+        let r = (fog.color.r() + 0.1 * delta).min(1.0);
+        fog.color.set_r(r);
+    }
+
+    if keycode.pressed(KeyCode::LBracket) {
+        let r = (fog.color.g() - 0.1 * delta).max(0.0);
```

Comment:
You should swap your variable names for the latter if statements to be less confusing :p

Reply:
Oops.